js/Home-page.js

```diff
@@ -0,0 +1,147 @@
+
+const sidebar = document.getElementById('sidebar')
+
+function toggleSidebar(){
+  sidebar.classList.toggle('close') 
+  toggleButton.classList.toggle('rotate')
+
+  closeAllSubMenus()
+}
+
+function toggleSubMenu(button){
+
+  if(!button.nextElementSibling.classList.contains('show')){
+    closeAllSubMenus()
+  }
+
+  button.nextElementSibling.classList.toggle('show')
+  button.classList.toggle('rotate')
+
+  if(sidebar.classList.contains('close')){
+    sidebar.classList.toggle('close')
+    toggleButton.classList.toggle('rotate')
+  }
+}
+
+function closeAllSubMenus(){
+  Array.from(sidebar.getElementsByClassName('show')).forEach(ul => {
+    ul.classList.remove('show')
+    ul.previousElementSibling.classList.remove('rotate')
+  })
+}
+
+
+
+const animals = {
+  Echidna: {
```

Comment:
function Animal(image, name, group, lifespan, weight, food, length, found, description) {
    this.image = image;
    this.name = name;
    this.group = group;
    this.lifespan = lifespan;
    this.weight = weight;
    this.food = food;
    this.length = length;
    this.found = found;
    this.description = description;
  }
  try like this.